docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,514 @@
+to discover the new public key. This rotation strategy could be useful for
+external services that do not automatically rediscover the OIDC issuer metadata
+when the token is signed with still unknown to them key.
+
+The Kubernetes API server extended by the Gardener API server is already issuing
+JWTs for the kubernetes service accounts. To completely separate workload
+identity JWTs from service accounts JWTs, Gardener API will accept issuer URL
+parameter which value should not be the same as the issuer of the kubernetes
+service accounts. The workload identity issuer url should not be among the
+accepted issuers of the Kubernetes API server. Other configuration options for
+the Gardener API server will be the private key used to sign the tokens, the
+minimal, maximal and the default durations for each token. The private key also
+should not be shared with the Kubernetes API server.
+
+When Gardener API server is using own issuer and signing keys, the service
+account token authenticator of the Kubernetes API server will not reject the
+workload identity JWTs because:
+
+- the issuer of the tokens is not accepted
+- the tokens are not signed by trusted key
+- workload identity JWTs are not referring to any kubernetes service account
+- Gardener API will not serve the purpose of authentication or authorization
+  webhook, it will also not implement any authentication or authorization based
+  on the workload identity JWTs, it will just generate and sign them.
+
+Gardener API server will use the global configurations, `WorkloadIdentity` and
+`TokenRequest` specifications to issue JSON Web Tokens. Later, if a use case is
+identified, it could feature custom claims in the `gardener.cloud` claim
+namespace that contain additional information about the context of use of the
+token, e.g. metadata about the shoot, seed, project, landscape, etc.
+
+A sample payload of a token will look like:
+
+```json
+{
+    "aud": [
+        "service-foo-provider-bar"
+    ],
+    "exp": 1707315742,
+    "iat": 1707312142,
+    "nbf": 1707312142,
+    "iss": "https://workload-identity.gardener-local.gardener.cloud",
+    "sub": "gardener.cloud:workloadidentity:<workloadidentity-namespace>:<workloadidentity-name>:<workloadidentity-uid>",
+    "gardener.cloud": {
+        "workloadidentity": {
+            "name": "<workloadidentity-name>",
+            "namespace": "<workloadidentity-namespace>",
+            "uid": "<workloadidentity-uid>",
+        },
+        "shoot": {
+            "name": "<shoot-name>",
+            "namespace": "<shoot-namespace>",
+            "uid": "<shoot-uid>",
+        },
+        "project": {
+            "name": "<project-name>",
+            "uid": "<project-uid>",
+        },
+        "seed": {
+            "name": "<seed-name>",
+            "uid": "<seed-uid>",
+        },
+        "landscape": {
+            "id": "<gardener-landscape-id>",
+        },
```

Comment:
What's this? Is this https://github.com/gardener/gardener/blob/07052b52a2d2eabb1a5991f1a4f3c92555abbb55/charts/gardener/controlplane/charts/runtime/templates/apiserver/deployment.yaml#L192? Perhaps it should be renamed accordingly?